services/web/server/src/simcore_service_webserver/licenses/_licensed_checkouts_api.py

```diff
@@ -0,0 +1,109 @@
+from aiohttp import web
+from models_library.api_schemas_resource_usage_tracker import (
+    licensed_items_checkouts as rut_licensed_items_checkouts,
+)
+from models_library.api_schemas_webserver import (
+    licensed_items_checkouts as webserver_licensed_items_checkouts,
+)
+from models_library.licensed_items import LicensedItemID
+from models_library.products import ProductName
+from models_library.resource_tracker import ServiceRunId
+from models_library.resource_tracker_licensed_items_checkouts import (
+    LicensedItemCheckoutID,
+)
+from models_library.users import UserID
+from models_library.wallets import WalletID
+from servicelib.rabbitmq.rpc_interfaces.resource_usage_tracker import (
+    licensed_items_checkouts,
+)
+
+from ..rabbitmq import get_rabbitmq_rpc_client
+from ..users.api import get_user
+from ..wallets.api import get_wallet_by_user
+
+
+async def checkout_licensed_item_for_wallet(
+    app: web.Application,
+    licensed_item_id: LicensedItemID,
+    wallet_id: WalletID,
+    product_name: ProductName,
+    num_of_seats: int,
+    service_run_id: ServiceRunId,
+    user_id: UserID,
+) -> webserver_licensed_items_checkouts.LicensedItemCheckoutGet:
```

Comment:
We should not contaminate service layer with controller models. 
For now i guess it can pass but I am writing something about it and we should refactor all these so we slowly decouple nicely all these layers.

Reply:
We need to discuss what is reasonable. Currently, we decouple the Repository and Controller layers nicely using two models, which requires some extra work but pays off in the long run. I usually connect these two in the service layer. Introducing yet another model between the service layer and the controller layer doesn't show immediate benefits to me at this point. However, if necessary, it can be done on a case-by-case basis. In any case, it is usually a one-to-one mapping.